be/src/storage/rowset/vectorized/segment_iterator.cpp

```diff
@@ -412,12 +434,16 @@ Status SegmentIterator::_get_row_ranges_by_keys() {
         rowid_t upper_rowid = num_rows();
 
         if (!range.upper().empty()) {
-            _init_column_iterators(range.lower().schema());
+            _push_init_ctxs();
```

Comment:
Why need push and pop init ctxs here?

Reply:
_init_column_iterators in _get_row_ranges_by_keys will modify _column_iterators, which depends by function _init_global_dict_decoder

Reply:
init context could be removed